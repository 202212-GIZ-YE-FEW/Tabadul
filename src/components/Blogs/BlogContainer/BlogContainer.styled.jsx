import styled from "@emotion/styled";

export const Header = styled.h1`
    text-align: center;
    font-style: normal;
    font-weight: 700;
    font-size: 40px;
    line-height: 84px;
    color: #3c4347;
    margin-bottom: 90px;
    @media screen and (max-width: 768px) {
        font-size: 30px;
    }
`;

export const Container = styled.div`
    display: flex;
    justify-content: center;
    gap: 50px;
    margin-bottom: 90px;
`;
