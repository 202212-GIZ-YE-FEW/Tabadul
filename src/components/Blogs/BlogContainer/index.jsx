import { useTranslation, withTranslation } from "next-i18next";
import React from "react";

import { CardLink } from "@/components/ListedItems/ListedItems.styled";
import { Line } from "@/components/PopularItems/PopularItems.styled";

import { Header } from "./BlogContainer.styled";
import SingleBlogCard from "../SingleBlogCard";
import { BlogCard } from "../SingleBlogCard/SingleBlogCard.styled";

const BlogContainer = ({ blogs }) => {
    const { t } = useTranslation("common");
    function translateMe(word) {
        return t(word);
    }

    return (
        <>
            <Header>{translateMe("blogs")}</Header>
            <BlogCard>
                {blogs?.slice(0, 4).map((blog) => {
                    return (
                        <CardLink key={blog.id} href={`Blogs/${blog.id}`}>
                            <SingleBlogCard
                                blogTitle={blog.title}
                                blog={blog.blog}
                                blogDate={blog.date}
                                blogImage={blog.image}
                            />
                        </CardLink>
                    );
                })}
            </BlogCard>
            <Line />
        </>
    );
};

export default BlogContainer;
