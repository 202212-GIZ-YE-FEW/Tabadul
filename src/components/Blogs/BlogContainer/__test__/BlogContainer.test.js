import renderer from "react-test-renderer";

import BlogContainer from "..";

it("renders correctly", () => {
    const tree = renderer.create(<BlogContainer />).toJSON();
    expect(tree).toMatchSnapshot();
});
