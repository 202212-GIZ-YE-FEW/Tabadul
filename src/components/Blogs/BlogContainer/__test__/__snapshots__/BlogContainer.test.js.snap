// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="css-gtmh19"
  >
    Blogs
  </h1>,
  <div
    className="css-wql7ev"
  >
    <div
      className="css-nicycs"
    >
      <div
        className="css-jd87c1"
        style={
          Object {
            "width": "280px",
          }
        }
      >
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-title--secondary grid-gap-10"
            >
              <div
                className="shimmer shimmer-title-line"
              />
              <div
                className="shimmer shimmer-title-line"
              />
            </div>
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div
        className="css-jd87c1"
        style={
          Object {
            "width": "280px",
          }
        }
      >
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-title--secondary grid-gap-10"
            >
              <div
                className="shimmer shimmer-title-line"
              />
              <div
                className="shimmer shimmer-title-line"
              />
            </div>
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div
        className="css-jd87c1"
        style={
          Object {
            "width": "280px",
          }
        }
      >
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-title--secondary grid-gap-10"
            >
              <div
                className="shimmer shimmer-title-line"
              />
              <div
                className="shimmer shimmer-title-line"
              />
            </div>
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div
        className="css-jd87c1"
        style={
          Object {
            "width": "280px",
          }
        }
      >
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-title--secondary grid-gap-10"
            >
              <div
                className="shimmer shimmer-title-line"
              />
              <div
                className="shimmer shimmer-title-line"
              />
            </div>
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <hr
    className="css-y5d9ts"
  />,
]
`;
